---
title: "class11"
author: 'Vince (PID: A15422556)'
date: "2/22/2022"
output:
  pdf_document: default
  html_document: default
---

> Q1. How many genes are in this dataset? 38694

> Q2. How many ‘control’ cell lines do we have? 4 'control' cell lines

```{r}
counts <- read.csv("airway_scaledcounts.csv", row.names=1)
metadata <-  read.csv("airway_metadata.csv")

nrow(counts)
sum(metadata$dex == "control")
```


Check to see that columns of countdata and coldata (metadata) match.

```{r}
all(metadata$id == colnames(counts))
```


> Q3. How would you make the above code in either approach more robust? See code below.

> Q4. Follow the same procedure for the treated samples (i.e. calculate the mean per gene across drug treated samples and assign to a labeled vector called treated.mean) See code below.


## Extract control and treated counts for comparison

Extract the control counts columns.

```{r}
control.ids <- metadata[metadata$dex == "control", "id"]
control.counts <- counts[,control.ids]

control.mean <- rowMeans(control.counts)
head(control.mean)
```


```{r}
treated.ids <- metadata[metadata$dex == "treated", "id"]
treated.counts <- counts[,treated.ids]

treated.mean <- rowMeans(treated.counts)
head(treated.mean)
```


Plot comparing treated vs. control.

> Q5 (a). Create a scatter plot showing the mean of the treated samples against the mean of the control samples. Your plot should look something like the following.

>Q5 (b).You could also use the ggplot2 package to make this figure producing the plot below. What geom_?() function would you use for this plot? geom_point()

```{r}
plot(treated.mean, control.mean,
     xlab = "log of drug treated counts",
     ylab = "log of control counts")
```


> Q6. Try plotting both axes on a log scale. What is the argument to plot() that allows you to do this? log="xy"

```{r}
plot(treated.mean, control.mean, log = "xy",
     xlab = "log of drug treated counts",
     ylab = "log of control counts")
```



Changes in gene expression: treated vs. control. This would represent points (i.e. genes) that do not lie on the diagonal.


```{r}
log2fc <- log2(treated.mean / control.mean)

meancounts <- data.frame(control.mean, treated.mean, log2fc)
head(meancounts)
```

> Q7. What is the purpose of the arr.ind argument in the which() function call above? Why would we then take the first column of the output and need to call the unique() function? Tells the row and column where the values are true.

```{r}
zero.vals <- which(meancounts[,1:2]==0, arr.ind=TRUE)

to.rm <- unique(zero.vals[,1])
mycounts <- meancounts[-to.rm,]
head(mycounts)

nrow(mycounts)
```


> Q8. Using the up.ind vector above can you determine how many up regulated genes we have at the greater than 2 fc level? 250

> Q9. Using the down.ind vector above can you determine how many down regulated genes we have at the greater than 2 fc level? 21503

> Q10. Do you trust these results? Why or why not? No, we need a p-value

"Up" genes

```{r}
sum(mycounts$log2fc > 2, na.rm = TRUE)
```


"Down" genes
```{r}
sum(mycounts$log2fc < 2, na.rm = TRUE)
```





Missing the stats (are differences significant):

# DESeq2 Analysis 

```{r}
library(DESeq2)
```


Package wants input in a specific way:

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex)
dds
```

Run the DESeq analysis.

```{r}
dds <- DESeq(dds)

```


Results

```{r}
res <- results(dds)
res
```



# Volcano Plot

```{r}
plot( res$log2FoldChange,  -log(res$padj), 
      xlab="Log2(FoldChange)",
      ylab="-Log(P-value)")
abline(h=-log(0.05), col="gray")
abline(v=c(-2,2), col="gray")
```


Add color to the plots

```{r}
mycols <- rep("gray", nrow(res))

mycols[res$padj < 0.005] <- "red"
mycols[abs(res$log2FoldChange) < 2] <- "gray"

plot(res$log2FoldChange, -log(res$padj), col=mycols)
```





---

# Adding annotation data

To help interpret our results we need to understand what the differentially expressed genes are. A first step is to get the gene names (i.e. gene SYMBOLs).

```{r}
library("AnnotationDbi")
library("org.Hs.eg.db")

```


What DB identifiers can I look up?

```{r}
columns(org.Hs.eg.db)
```


Use `mapIds()` function to translate between different IDs.

```{r}
res$symbol <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",        # The format of our genenames
                     column="SYMBOL",          # The new format we want to add
                     multiVals="first")

head(res)
```


> Q11. Run the mapIds() function two more times to add the Entrez ID and UniProt accession and GENENAME as new columns called res$entrez, res$uniprot and res$genename. See code below.

```{r}
res$entrez <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",        # The format of our genenames
                     column="ENTREZID",          # The new format we want to add
                     multiVals="first")

res$uniprot <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",        # The format of our genenames
                     column="UNIPROT",          # The new format we want to add
                     multiVals="first")

res$genename <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",        # The format of our genenames
                     column="GENENAME",          # The new format we want to add
                     multiVals="first")

head(res)
```





# Pathway analysis with R and Bioconductor

> Q12. Can you do the same procedure as above to plot the pathview figures for the top 2 down-reguled pathways? See code below.

```{r}
library(pathview)
library(gage)
library(gageData)

data(kegg.sets.hs)

# Examine the first 2 pathways in this kegg set for humans
head(kegg.sets.hs, 2)
```


Need a vector of fold-change labeled with the names of our genes in ENTREZ format.

```{r}
foldchanges = res$log2FoldChange
names(foldchanges) = res$entrez
head(foldchanges)
```


Run the GAGE analysis passing in our foldchange vector and KEGG genesets we are interested in.

```{r}
# Get the results
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```


Look at what is contained in this `keggres` results object (i.e. its attributes).

```{r}
attributes(keggres)
```

```{r}
# Look at the first three down (less) pathways
head(keggres$less, 3)
```


Map my results onto any KEGG pathway. Do ths manually first by selecting one of the pathway IDs from above.

```{r}
pathview(gene.data=foldchanges, pathway.id="hsa05310")

```


![](hsa05310.pathview.png)

## Final step is to save our results.

```{r}
write.csv(res, file="deseq_results.csv")
```

